proto/spacemesh/smesher_types.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+package spacemesh;
+
+// User provided post data
+message CommitmentUserData {
+    string path = 1;                // Path to post file(s) on a volume
+    uint64 commitment_size = 2;     // User specified commitment size in bytes
+    uint32 processor_id = 3;        // Processor to be used for smeshing. e.g. cpu, gpu0, gpu1, etc...
+    bool   append = 4;              // Append to existing file if it exists. Otherwise - overwrite
+    bool   throttle = 5;            // Throttle down init phase computations while user is interactive on system
+}
+
+message CommitmentProcessor {
+    uint32 id = 1;                  // Numerical identifier - unique per system
+    string processor_name = 2;      // Proc info, e.g. Nvidia `Geforce RTX`, `AMD Radeon`, `Intel Genuine i7 CPU`
+    string api = 3;                 // Pprocessing api to be used with the processor. e.g. `Vulkan`, `Cuda` or `x86-64`
+}
+
+message CommitmentProcessors {
+    repeated CommitmentProcessor processors = 1;
+}
+
+message DeleteCommitment {
+    bool delete_commitment = 1;
+}
+
+// Proof of space commitment data
+message CommitmentData {
+    enum FileStatus {
+        NOT_FOUND = 0;  // Commitment file was not initialized (in this or in previous node sessions)
```

Comment:
Please add `UNDEFINED = 0` as first element in enum (required by protobuf)

Reply:
Are you 100% sure about this. The field in the message can only have one of these 3 possible statues and should never return UNDEFINED. Looking at the protobuf docs - they show enums without UNDEFINED = 0 as the first element... where in the docs do you see this is required?

Reply:
This comes from the [protobuf style guide](https://developers.google.com/protocol-buffers/docs/style#enums):

> Each enum value should end with a semicolon, not a comma. Prefer prefixing enum values instead of surrounding them in an enclosing message. The zero value enum should have the suffix UNSPECIFIED.

See also https://buf.build/docs/lint-checkers#enum_zero_value_suffix.

As I said on Slack, I think the most important thing is that we pick a convention and stick to it. I'm inclined to stick to these best practice conventions unless we have a compelling reason to deviate.

Let's move this thread to #17.